Fully working version

madeye-matt opened the pull request:
- supports including and excluding of tasks from the command line
- supports multiple environments via the --environment directive

Apologies, I accidentally pushed the first batch of changes straight into master so this PR is not complete.   This repo will completely supersede rm-tools/collex-loader.  Once this has been approved, I will update the acceptance tests to check out *this* repo and run 'make env_test'

If you can think of anyone else who should review this, feel free to add them

madeye-matt committed:
Fully working version

- supports including and excluding of tasks from the command line
- supports multiple environments via the --environment directive

benjefferies commented on tasks.py:
```diff
@@ -32,20 +47,91 @@ def is_task_directory(name):
-    subprocess.run(["make", "-f", MAKEFILE_NAME], cwd=task_directory)
+    subprocess.run(["make", "-f", MAKEFILE_NAME], cwd=task_directory, env=env)
+
+def parse_environment_file_line(line):
```
Might be worth looking at https://github.com/pypa/pipenv/blob/master/docs/advanced.rst#-automatic-loading-of-env. The whole env parsing would be redundant if we could use it

madeye-matt replied:
Yes - should be able to say change default make target to:
```
PIPENV_DOTENV_LOCATION=environment.production pipenv run python tasks.py
```
and test target similarly.  Will try it now - also I have a way to specify the make target from an env variable so should be able to run different command lines in docker/cf

benjefferies reviewed (commented):
Looks good. Will run it and check its all working against docker

madeye-matt committed:
Added facility to override make targets with environment variables

madeye-matt committed:
Removed manual processing of env to rely on pipenv

madeye-matt committed:
Added initialisation for comms template service plus refactoring

- Added python script for batching http requests
- Added http requests for initialising comms template
  service (classfiers and templates)
- Removed pipfile etc from task directories
- Created lib directory for common code

madeye-matt committed:
Included full set of comms template records

madeye-matt committed:
Work to replicate exactly the actions of the acceptance tests 'make setup'

madeye-matt committed:
Updated README to include creating database & http tasks

Also removed old spurious version of batch HTTP uploader
